### ISSUE TITLE
package request: wezterm

### ISSUE THREAD

**Issue by colemickens** [NixOS/nixpkgs#90696]
**Project description**
It's another GPU-accelerated terminal emulator written in Rust. Seems natural to include in nixpkgs.

https://github.com/wez/wezterm

**Metadata**

* homepage URL: https://github.com/wez/wezterm
* source URL: https://github.com/wez/wezterm
* license: `MIT`
* platforms: `unix`

**Comment by lucc** [comment]
I recently tried to build it following the [docs](https://wezfurlong.org/wezterm/install/source.html).  I tried to figure out how the dependencies are called on nixos.  The [`get-deps` script](https://github.com/wez/wezterm/blob/master/get-deps) is only for other distros.

This is my attempt 
```
nix-shell --run "cargo build --release" -p cargo gcc cmake dbus fontconfig pkgconf python3 wayland libxkbcommon xorg.libX11 xorg.xcbutilkeysyms xorg.xcbutilwm dbus_libs
```
but it fails when trying to compile the [x11 crate](https://crates.io/crates/x11). Do you know which package I need to use on nixos to compile that?

**Comment by bew** [comment]
There is some discussion on wezterm's gitter: https://gitter.im/wezterm/Lobby?at=5fada3f5b4283c208a5f7d5c

Here is a recap':
I tried to build it as well a few days ago:
The standard `buildRustPackage` helper does something wonky with the packages and I get a version error with some dependencies.
Building the latest stable release manually with cargo doesn't work either for a different reason. The rust version of nixos 20.09 is too old (v1.45 vs v1.47), the build has been fixed in master for rust 1.45.
I tried to update to rust 1.47, but I got the same error. I think I did something wrong (still relatively new to Nix).
 
I tried to compile the latest master with rust 1.45, it works, but then when I run it I get an error saying it can't find `libEGL.so`.
And I stopped searching for now. If anyone wants to continue, feel free!

**Comment by steveeJ** [comment]
I'm in the process of packaging wezterm and have succeeded to build and run it from a nix-shell environment. However, packaging it is proving difficult due to issues with the dependency vendoring that the `rustPlatform.buildRustPackage` entails. I reproduced using `cargo vendor` alone and created a ticket upstream so we can get this resolved: https://github.com/wez/wezterm/issues/337
